### PR TITLE
Upload files

### PR DESCRIPTION
**Pull request by joaojmendes**
#### What's in this Pull Request?

New Control Upload Files

### CODE REVIEW

File: src/controls/uploadFiles/components/DocumentList/DocumentList.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from 'react';
+
+import strings from 'ControlStrings';
+import { useAtom } from 'jotai';
+import {
+  isEqual,
+  pullAllWith,
+} from 'lodash';
+import {
```
Comment (joelfmrodrigues) [inline src/controls/uploadFiles/components/DocumentList/DocumentList.tsx:11]: Can you please import from office-ui-fabric-react/lib/xxxxx

File: src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    (props) => {
+      return (
+        <ActionButton
+          iconProps={{ iconName: "Delete" , styles: buttonIconStyles}}
+          styles={commandbarButtonStyles}
+          onClick={() => {
+            setIsAllSelected(false);
+            onDelete();
+          }}
+          text="Delete"
```
Comment (joelfmrodrigues) [inline src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx:51]: Can you please use a loc string

File: src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        };
+      });
+    }
+  }, [isAllSelected, files]);
+
+  const commanbarSelectAllButton: IComponentAs<ICommandBarItemProps> = React.useCallback(
+    (props) => {
+      return (
+        <Stack horizontal verticalAlign="center" horizontalAlign="start" tokens={{ childrenGap: 15 }}>
+          {files.length > 0 ? ( <Checkbox styles={checkBoxStyles} label="Select all" checked={isAllSelected} onChange={
```
Comment (joelfmrodrigues) [inline src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx:80]: Can you please use a loc string

File: src/controls/uploadFiles/components/UploadFiles/useUploadFilesStyles.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { useAtomValue } from 'jotai/utils';
+import { FontWeights } from 'office-ui-fabric-react';
```
Comment (joelfmrodrigues) [inline src/controls/uploadFiles/components/UploadFiles/useUploadFilesStyles.ts:5]: Can this use a more specific import?

File: src/loc/en-us.ts [hunk 5]
```diff
@@ -406,5 +406,9 @@ define([], () => {
     ModernAudioDecVol: "Decrease Volume",
     ModernAudioMute: "Mute",
     ProgressStepsIndicatorNoSteps: "No steps",
+    UploadFilesButtonLabel: "Add",
+    UpLoadFilesDragDropLabel: "Drag and Drop files or select Add File button",
+    UploadFilesUploadButtonLabel: "Upload",
```
Comment (joelfmrodrigues) [inline src/loc/en-us.ts:411]: This entry is missing on most loc files

File: src/loc/mystrings.d.ts [hunk 6]
```diff
@@ -409,6 +412,8 @@ declare interface IDateTimeStrings {
   L_RelativeDateTime_XDaysFutureIntervals: string;
   L_RelativeDateTime_XDaysIntervals: string;
   L_RelativeDateTime_Today: string;
+UploadFilesButtonLabel: string;
```
Comment (joelfmrodrigues) [inline src/loc/mystrings.d.ts:415]: Duplicated in line 2 and 4

File: src/webparts/controlsTest/ControlsTestWebPart.ts [hunk 7]
```diff
@@ -1,30 +1,31 @@
-import * as React from "react";
-import * as ReactDom from "react-dom";
+import * as React from 'react';
```
Comment (joelfmrodrigues) [inline src/webparts/controlsTest/ControlsTestWebPart.ts:1]: Can you try to revert the formatting changes?

File: src/webparts/controlsTest/components/ControlsTest.tsx [hunk 8]
```diff
@@ -1,197 +1,218 @@
...
-import {
-  Stack,
-} from "office-ui-fabric-react/lib/Stack";
-import {
-  Text,
-} from "office-ui-fabric-react/lib/Text";
-import {
-  TextField
-} from "office-ui-fabric-react/lib/TextField";
+import * as React from 'react';
```
Comment (joelfmrodrigues) [inline src/webparts/controlsTest/components/ControlsTest.tsx:1]: Can you try to revert the formatting changes?

File: src/webparts/controlsTest/components/TestControl.tsx [hunk 9]
```diff
@@ -1,37 +1,39 @@
-import * as React from "react";
+import * as React from 'react';
```
Comment (joelfmrodrigues) [inline src/webparts/controlsTest/components/TestControl.tsx:1]: Can you try to revert the formatting changes?

**Review by joelfmrodrigues — CHANGES_REQUESTED** [review]

File: src/controls/uploadFiles/components/DocumentList/DocumentList.tsx (hunk 1 above)
Comment (joaojmendes) [reply src/controls/uploadFiles/components/DocumentList/DocumentList.tsx:11]: done

File: src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx (hunk 2 above)
Comment (joaojmendes) [reply src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx:51]: done

File: src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx (hunk 3 above)
Comment (joaojmendes) [reply src/controls/uploadFiles/components/FileCommandBar/FileCommandBar.tsx:80]: done

File: src/controls/uploadFiles/components/UploadFiles/useUploadFilesStyles.ts (hunk 4 above)
Comment (joaojmendes) [reply src/controls/uploadFiles/components/UploadFiles/useUploadFilesStyles.ts:5]: done

File: src/loc/en-us.ts (hunk 5 above)
Comment (joaojmendes) [reply src/loc/en-us.ts:411]: done

File: src/loc/mystrings.d.ts (hunk 6 above)
Comment (joaojmendes) [reply src/loc/mystrings.d.ts:415]: done

File: src/webparts/controlsTest/components/ControlsTest.tsx (hunk 8 above)
Comment (joaojmendes) [reply src/webparts/controlsTest/components/ControlsTest.tsx:1]: done

File: src/webparts/controlsTest/components/TestControl.tsx (hunk 9 above)
Comment (joaojmendes) [reply src/webparts/controlsTest/components/TestControl.tsx:1]: done

File: src/webparts/controlsTest/ControlsTestWebPart.ts (hunk 7 above)
Comment (joaojmendes) [reply src/webparts/controlsTest/ControlsTestWebPart.ts:1]: done

**Comment by joelfmrodrigues** [comment]
@joaojmendes many thanks! I have addressed the issues locally and merged the PR

**Pull request merged**